File: finder4j-backtrack-examples/src/test/java/com/github/K0zka/finder4j/backtrack/examples/queens/QueensStepTest.java
package com.github.K0zka.finder4j.backtrack.examples.queens;


import org.junit.Assert;
import org.junit.Test;

import com.github.K0zka.finder4j.backtrack.examples.queens.QueensState;
import com.github.K0zka.finder4j.backtrack.examples.queens.QueensStep;

public class QueensStepTest {
	@Test
	public void take() {
		QueensState state = new QueensState(new Short[]{1});
		QueensStep step = new QueensStep((short)0, (short)2);
		QueensState newState = step.take(state);
		Assert.assertEquals(2, newState.getQueens()[0].shortValue());
	}
}

File: finder4j-backtrack-examples/src/test/kotlin/com/github/k0zka/finder4j/backtrack/examples/queens/QueensStepTest.kt
package com.github.k0zka.finder4j.backtrack.examples.queens


import org.junit.Assert
import org.junit.Test

import com.github.k0zka.finder4j.backtrack.examples.queens.QueensState
import com.github.k0zka.finder4j.backtrack.examples.queens.QueensStep

class QueensStepTest {
	@Test
	fun take() {
		val state = QueensState(arrayOf(1.toShort()).toShortArray())
		val step = QueensStep(0.toShort(), 2.toShort())
		val newState = step.take(state)
		Assert.assertEquals(2, newState.queens[0].toShort().toLong())
	}
}
